### PR TITLE
For automation, we need to publish rule templates

### PR DESCRIPTION
**Pull request by geraldguillaume**
Signed-off-by: Gerald Guillaume <geraldguillaume@eaton.com>
GET /api/v1/alerts/templates?type=all
returns
[
{
"name" :"average.humidity@__datacenter__.rule"
"template" : {
    "threshold" : {
        "rule_name"     :   "average.humidity@__name__",
        "rule_desc"     :   "Humidity",
        "rule_class"    :   "Humidity in data center",
        "rule_hierarchy" :   "environment.humidity.dc",
        "target"        :   "average.humidity@__name__",
        "element"       :   "__name__",
        "values_unit"   :   "%",
        "values"        :   [ {"low_warning"   : "40"},
                              {"low_critical"  : "30"},
                              {"high_warning"  : "60"},
                              {"high_critical" : "70"} ],
        "results"       :   [
            { "low_warning"   : { "action" : [{ "action": "EMAIL" },{ "action": "SMS" }], "severity" : "WARNING", "description" : "Humidity in datacenter is low" }},
            { "low_critical"  : { "action" : [{ "action": "EMAIL" },{ "action": "SMS" }], "severity" : "CRITICAL", "description" : "Humidity in datacenter is critically low" }},
            { "high_warning"  : { "action" : [{ "action": "EMAIL" },{ "action": "SMS" }], "severity" : "WARNING", "description" : "Humidity in datacenter is high" }},
            { "high_critical" : { "action" : [{ "action": "EMAIL" },{ "action": "SMS" }], "severity" : "CRITICAL", "description" : "Humidity in datacenter is critically high" }} ]
    }
},
"asset_list" : "datacenter-3"
}
]

### CODE REVIEW

**Comment by EldoreiJK** [comment]
Note that you won't get flexible rules probably (at least function that has been done on fty-alert-engine list-rules didn't return them), and you won't get templates for fty-nut related alerts.

**Comment by EldoreiJK** [comment]
Also licensing template is hardcoded in fty-alert-engine

**Comment by geraldguillaume** [comment]
@EldoreiJK : flexible templates are present in  /usr/share/bios/fty-autoconfig

File: src/web/src/alert_templates.ecpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    zstr_free (&result);
+    zmsg_destroy (&msg);
+    http_die ("internal-error", "LIST template bad response");
+}
+zstr_free (&result);
+std::string json ="[";
+char* template_name = zmsg_popstr (msg);
+char* templat = zmsg_popstr (msg);
+char* asset_list = zmsg_popstr (msg);
+while(template_name){
```
Comment (barraudl) [inline src/web/src/alert_templates.ecpp]: Need to be rework using cxxtools json constructor

**Pull request merged**